### DISCUSSION TITLE
Make `react-select` auto-select if only 1 option and no default value

### DISCUSSION THREAD

**Discussion by christianJessMark** [react-hook-form/react-hook-form#6159] [Q&A] [answered]
I'm using (controlled) `react-select` with `reacht-hook-form`. I'm coming from `react-final-form` where I had the possibility to make my selects automatically select an option, if only one option is available. I've struggled a long time, trying to do the same with `react-hook-form`. 

```
<Controller
  name="ReactSelect"
  isClearable
  control={control}
  render={({ field }) => (
    <ReactSelect
      {...field}
      options={[
        { value: "vanilla", label: "Vanilla" }
      ]}
    />
  )}
/>
 ```
 
 So in the example above from [official examples](https://codesandbox.io/s/react-hook-form-v7-controller-5h1q5?file=/src/index.js), I've removed the default value and want to have the select automatically select "Vanilla" - is that possible?

**Comment by bluebill1049** [comment, marked as answer]
You will need to provide defaultValues at useForm or defaultValue at useController.

**Reply by christianJessMark** [reply]
I see it works in the example, but not in my local setup :-( 

That is, it works if I render the form with only one option. But if I need to dynamically change the options, dependent on another field, it doesn't work. 

Can the defaultValue be updated dynamically? And is it ok to mix inline default values with default values on the form level?

**Reply by bluebill1049** [reply]
https://react-hook-form.com/api/useform/setvalue
https://react-hook-form.com/api/useform/reset

**Reply by christianJessMark** [reply]
`setValue` work, though I have to wrap it in a `useEffect` to detect if the number of options change. Would it be worth considering having `defaultValue` re-render the field if it changes?

**Reply by bluebill1049** [reply]
`defaultValue` is not a local state, this is one of the unique things about react hook form. we don't trigger re-render on your entire form (there is no form state), while let user/you set the input value. defaultValue is only the initial state of the input, it doesn't sync with what user is updated.

**Reply by christianJessMark** [reply]
Fair point! It takes a lot of different thinking coming from a library with form state to react hook form. Thanks for the answers!
